EngineLayer/ModernSearch/ModernSearchEngine.cs

```diff
@@ -118,45 +144,38 @@ protected override MetaMorpheusEngineResults RunSpecific()
+                            if (bestPeptides[j] != null)
                             {
-                                int notch = searchMode.Accepts(thisScanprecursorMass, candidatePeptide.MonoisotopicMassIncludingFixedMods);
-                                if (notch >= 0)
+                                newPsms[j][i] = new Psm(bestPeptides[j][0], bestNotches[j][0], bestScores[j], i, thisScan);
+                                for (int k = 1; k < bestPeptides[j].Count; k++)
                                 {
-                                    bestPeptides[j] = new List<CompactPeptide> { candidatePeptide };
-                                    bestScores[j] = consideredScore;
-                                    bestNotches[j] = new List<int> { notch };
+                                    newPsms[j][i].Add(bestPeptides[j][k], bestNotches[j][k]);
                                 }
                             }
                         }
                     }
-                    for (int j = 0; j < searchModesCount; j++)
+                    lock (outputObject)
                     {
-                        if (bestPeptides[j] != null)
+                        scansSeen += fff.Item2 - fff.Item1;
+                        var new_progress = (int)((double)scansSeen / (listOfSortedms2ScansLength) * 100);
+                        if (new_progress > old_progress)
                         {
-                            newPsms[j][i] = new Psm(bestPeptides[j][0], bestNotches[j][0], bestScores[j], i, thisScan);
-                            for (int k = 1; k < bestPeptides[j].Count; k++)
-                            {
-                                newPsms[j][i].Add(bestPeptides[j][k], bestNotches[j][k]);
-                            }
+                            ReportProgress(new ProgressEventArgs(new_progress, "In modern search loop", nestedIds));
+                            old_progress = new_progress;
                         }
                     }
-                }
-                lock (outputObject)
-                {
-                    scansSeen += fff.Item2 - fff.Item1;
-                    var new_progress = (int)((double)scansSeen / (listOfSortedms2ScansLength) * 100);
-                    if (new_progress > old_progress)
-                    {
-                        ReportProgress(new ProgressEventArgs(new_progress, "In modern search loop", nestedIds));
-                        old_progress = new_progress;
-                    }
-                }
-            });
-            return new SearchResults(newPsms, this);
+                });
+                return new SearchResults(newPsms, this);
+            }
+            else
+            {
+                NonSpecificEnzymeSearch.NonSpecificEnzymeEngine NSEE = new NonSpecificEnzymeSearch.NonSpecificEnzymeEngine(listOfSortedms2Scans, peptideIndex, keys, fragmentIndex, fragmentTolerance, searchModes, nestedIds, addCompIons, lp, protease, minPeptideLength);
```

Comment:
Same here. Instead of calling an engine from an engine, put the if statement in the search task